### PR TITLE
Successful metadata queries not logged

### PR DESCRIPTION
**Pull request by highker**
Two state machines are used in class QueryStateMachine: queryState and finalQueryInfo. The query log callback function was previously hooked only with finalQueryInfo, which does not have information with the query state (RUNNING, FINISHING, etc). The state of finalQueryInfo will be updated only when the output result is handled. This is not true for metadata queries, which do not have outputs. This patch hooks the metadata query callback function with queryState. The callback is invoked when the query finishes.

### CODE REVIEW

File: presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.facebook.presto.execution.QueryManager;
+import com.facebook.presto.tests.DistributedQueryRunner;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.testng.annotations.Test;
+
+import static java.lang.String.format;
+
+public class TestRaptorQueryLogging
```
Comment (electrum) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java]: This test isn't specific to Raptor and should not go here. Move to `AbstractTestDistributedQueries`.

File: presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{
+    @Test
+    public void testQueryLoggingCount()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = RaptorQueryRunner.createRaptorQueryRunner(ImmutableMap.of(), false, false)) {
+            queryRunner.execute("CREATE TABLE foo (x bigint)");
+            queryRunner.execute("SELECT * FROM foo");
+            queryRunner.execute("DROP TABLE foo");
+            try {
```
Comment (electrum) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java]: Remove this try/catch. It shouldn't throw, and if it does, we definitely do not want to ignore it.

File: presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                queryRunner.execute("SELECT * FROM foo");
+            }
+            catch (Throwable e) {
+            }
+
+            QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
+            long queryCount = queryManager.getStats().getCompletedQueries().getTotalCount();
+            if (queryCount != 4) {
```
Comment (electrum) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java]: Use `assertEquals`, as it's simpler and provides a better error message on failure

File: presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor;
+
+import com.facebook.presto.execution.QueryManager;
+import com.facebook.presto.tests.DistributedQueryRunner;
+
```
Comment (electrum) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java]: The imports should not have all these extra blank lines. Please run code formatter.

File: presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java [hunk 5]
```diff
@@ -367,15 +367,15 @@ public QueryInfo createQuery(SessionSupplier sessionSupplier, String query)
         queryMonitor.queryCreatedEvent(queryInfo);
 
         queryExecution.addFinalQueryInfoListener(finalQueryInfo -> {
-                try {
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java]: This seems to be an unrelated formatting change.

File: presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java (hunk 2 above)
Comment (highker) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorQueryLogging.java]: Isn't the last statement going to trigger something like this:

com.facebook.presto.sql.analyzer.SemanticException: line 1:15: Table raptor.tpch.foo does not exist

	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitTable(StatementAnalyzer.java:698)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitTable(StatementAnalyzer.java:188)
	at com.facebook.presto.sql.tree.Table.accept(Table.java:53)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.analyzeFrom(StatementAnalyzer.java:1665)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuerySpecification(StatementAnalyzer.java:796)
	at com.facebook.presto.sql.analyzer.StatementAnalyzer.visitQuerySpecification(StatementAnalyzer.java:188)
	at com.facebook.presto.sql.tree.QuerySpecification.accept(QuerySpecification.java:127)

**Comment by highker** [comment]
ping

**Comment by haozhun** [comment]
@highker, you did good investigation in this pull request. However, the fix you had in this pull request does not fix the root of the problem. You exceeded my expectation on this pull request.

I'll close this now that I had put up a new pull request that supersedes this one.

Thank you!
